src/controls/dropdown-nav-menu-item.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import OutsideClickHandler from 'react-outside-click-handler'
+import { Link } from 'react-router'
+import classnames from 'classnames'
+
+const propTypes = {
+  name: PropTypes.string.isRequired,
+  submenuId: PropTypes.string.isRequired,
+  path: PropTypes.string.isRequired,
+  isExternalPath: PropTypes.bool.isRequired,
+  isSubmenuOpen: PropTypes.bool,
+  closeDesktopSubmenus: PropTypes.func.isRequired,
+  toggleSubmenu: PropTypes.func.isRequired,
+  isFirstItem: PropTypes.bool.isRequired,
+  children: PropTypes.node,
+  hideSubmenuButtonBeforeFocus: PropTypes.bool.isRequired,
+}
+
+const defaultProps = {
+  isSubmenuOpen: false,
+  children: null,
+}
+
+function DropdownNavMenuItem({
+  name,
+  submenuId,
+  path,
+  isExternalPath,
+  isSubmenuOpen,
+  closeDesktopSubmenus,
+  toggleSubmenu,
+  isFirstItem,
+  children,
+  hideSubmenuButtonBeforeFocus,
+}) {
+  // show submenu button on desktop, will always be shown on mobile
+  const [showSubmenuButton, setShowSubmenuButton] = useState(
+    !hideSubmenuButtonBeforeFocus
+  )
+  const menuItemOnFocus = () => {
+    if (hideSubmenuButtonBeforeFocus) setShowSubmenuButton(true)
+    closeDesktopSubmenus()
+  }
+  const menuItemOnBlur = () => {
+    if (hideSubmenuButtonBeforeFocus) {
+      // timeout needed to move from link to button without it disappearing
+      setTimeout(() => setShowSubmenuButton(false), 0)
+    }
+  }
+  const menuItemProps = {
+    onFocus: menuItemOnFocus,
+    onBlur: menuItemOnBlur,
+    role: 'menuItem',
+  }
+
+  return (
+    <li
+      className={classnames('menu-item parent-menu', {
+        'open-submenu': isSubmenuOpen,
+      })}
+      role="none"
+    >
+      <OutsideClickHandler onOutsideClick={closeDesktopSubmenus}>
```

Comment:
Similar question as before - are there use cases for closing all submenus vs. just the relevant submenu? Or should all submenus always be closed here?

If only one submenu can be open at a time, does that suggest that the value captured should be `submenuId` and not `submenuIds`?